lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info = {})
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if self.services.include?("Vehicle Registration") && vehicle.antique? == true
```

Comment:
I wouldn't call `self.services` use the instance variable from your `initialize`.  Also you will only need to check that `@services.include?("Vehicle Registration")` once and then you can proceed to check the `plate_type` for each scenario. An example:
```ruby
def register_vehicle(vehicle)
  return nil if !services.include?("Vehicle Registration")
  @registered_vehicles << vehicle
  if vehicle.antique? 
    @collected_fees += 25
    vehicle.plate_type = :antique
  elsif vehicle.electric_vehicle?
    @collected_fees += 200
    vehicle.plate_type = :ev
  else 
    @collected_fees += 100
    vehicle.plate_type = :regular 
  end
    @collected_fees
end
```
Something like that.